app/src/main/java/com/courseproject/tindar/ChatRecyclerViewAdapter.java

```diff
@@ -0,0 +1,144 @@
+    to future features like file-type messages.
+    In an ideal world, I'd refactor such that each case handles itself,
+    but I'm not currently familiar enough with all the inner workings here
+    and we have other more important features to focus on. Some things to potentially fix:
+    - constants for switch/if-else statements or remove them entirely
+ */
+
+
+/**
+ * I think this feeds into the RecyclerView for the one-on-one chat messages.
+ * Still very confused about what it actually does.
+ * @author Sophia Wan
+ */
+public class ChatRecyclerViewAdapter
+        extends RecyclerView.Adapter<ChatRecyclerViewAdapter.TindarMessageViewHolder>{
+    /**
+     * userID of current user. Needed check whether a message was sent or received.
+     * Might later refactor such that we don't need this.
+     */
+    private final String _userID;
+    /** List of messages to be displayed. Misnomer, but private so it doesn't really matter. */
+    private ArrayList<MessageModel> _displayedMessages;
+
+    /**
+     * Constructor. ChatRecyclerViewAdapter is intended to be an adapter for a view.
+     * @param displayedMessages list of messages that this adapter will help display
+     * @param userID User ID of the current user using the app
+     */
+    public ChatRecyclerViewAdapter(ArrayList<MessageModel> displayedMessages,
+                                   String userID){
+        this._displayedMessages = displayedMessages;
+        this._userID = userID;
+    }
+
+    /**
+     * Creates and adds TindarMessageViewHolder objects to the parent ViewGroup
+     * @param parent   The ViewGroup into which the new View will be added after it is bound to
+     *                 an adapter position.
+     * @param viewType The view type of the new View.
+     * @return
+     */
+    @NonNull
+    @Override
+    public TindarMessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        /*
+        this is where we will inflate the layout;
+        take the layout description from the XML
+        and populate the view hierarchy with actual View objects.
+        TODO: refactor to avoid switch statements
+         */
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View view;
+        switch(viewType){
+            case 0:
+                view = inflater.inflate(R.layout.recycler_row_message_sent,
+                        parent, false);
+                return new TindarMessageViewHolder(view);
+            case 1:
+                view = inflater.inflate(R.layout.recycler_row_message_received,
+                        parent, false);
+                return new TindarMessageViewHolder(view);
+        }
+        return null;
+    }
+
+    /**
+     * Updates a message TindarMessageViewHolder with the relevant details
+     * from the corresponding MessageModel
+     * @param holder   The ViewHolder which should be updated to represent the contents of the
```

Comment:
can item be more descriptive of what its referring to

Reply:
done